lib/philomena/tags.ex

```diff
@@ -39,43 +81,72 @@ defmodule Philomena.Tags do
   """
   @spec get_or_create_tags(String.t()) :: list()
   def get_or_create_tags(tag_list) do
-    tag_names = Tag.parse_tag_list(tag_list)
-
-    existent_tags =
-      Tag
-      |> where([t], t.name in ^tag_names)
-      |> preload([:implied_tags, aliased_tag: :implied_tags])
-      |> Repo.all()
-      |> Enum.uniq_by(& &1.name)
-
-    existent_tag_names =
-      existent_tags
-      |> Map.new(&{&1.name, true})
+    case Tag.parse_tag_list(tag_list) do
+      [] -> []
+      tag_names -> get_or_create_non_empty_tags_list(tag_names)
+    end
+  end
 
-    nonexistent_tag_names =
+  @spec get_or_create_non_empty_tags_list(list(String.t())) :: list()
+  defp get_or_create_non_empty_tags_list(tag_names) do
+    tags =
       tag_names
-      |> Enum.reject(&existent_tag_names[&1])
-
-    # Now get rid of the aliases
-    existent_tags =
-      existent_tags
-      |> Enum.map(&(&1.aliased_tag || &1))
-
-    new_tags =
-      nonexistent_tag_names
-      |> Enum.map(fn name ->
-        {:ok, tag} =
-          %Tag{}
-          |> Tag.creation_changeset(%{name: name})
-          |> Repo.insert()
-
-        %{tag | implied_tags: []}
-      end)
+      |> Enum.sort()
+      |> Enum.dedup()
+      |> Enum.map(
+        &(%Tag{}
```

Comment:
i'd not use this shorthand form here for clarity's sake, and instead would write `Enum.map(fn tag_name -> ...`